cgi-bin/LJ/CleanHTML.pm

```diff
@@ -1215,12 +1201,10 @@ sub clean
     # don't close tags that don't need a closing tag -- otherwise,
     # we output the closing tags in the wrong place (eg, a </td>
     # after the <table> was closed) causing unnecessary problems
-    if (ref $opts->{'autoclose'} eq "ARRAY") {
-        foreach my $tag (@{$opts->{'autoclose'}}) {
-            next if $tag =~ /^(?:tr|td|th|tbody|thead|tfoot|li)$/;
-            if ($opencount{$tag}) {
-                $newdata .= "</$tag>" x $opencount{$tag};
-            }
+    foreach my $tag ( @tagstack ) {
+        next if $tag =~ $slashclose_tags;
+        if ($opencount{$tag}) {
+            $newdata .= "</$tag>" x $opencount{$tag};
```

Comment:
We probably only want to output one close tag for each tag on the stack.

Example for both cases:

    <h1><h2><h3><h1><h2><h3>

Outputs:

     <h1><h2><h3><h1><h2><h3></h1></h1></h2></h2></h3></h3></h1></h1></h2></h2></h3></h3>